wagtail/admin/templates/wagtailadmin/shared/page_status_tag.html

```diff
@@ -1,18 +1,15 @@
-{% load i18n %}
+{% load i18n wagtailadmin_tags wagtailcore_tags %}
 {% if page.live %}
-    {% with page_url=page.url %}
-        {% if page_url is not None %}
-            <a href="{{ page_url }}" target="_blank" rel="noreferrer" class="status-tag primary" title="{% trans 'Visit the live page' %}">
-                <span class="visuallyhidden">{% trans "Current page status:" %}</span> {{ page.status_string }}
-            </a>
-        {% else %}
-            <span class="status-tag primary">
-                <span class="visuallyhidden">{% trans "Current page status:" %}</span> {{ page.status_string }}
-            </span>
-        {% endif %}
-    {% endwith %}
+    {% pageurl page as page_url %}
+    {% if page_url is not None %}
+        {% trans 'Visit the live page' as status_title %}
+        {% trans "Current page status:" as status_hidden_label %}
+        {% status page.status_string url=page_url title=status_title hidden_label=status_hidden_label classname="primary" attrs='target="_blank" rel="noreferrer"' %}
```

Comment:
@salty-ivy  I have refactored your original approach a bit, the `url` usage was not always needing to set target/rel so I added the ability to pass through an `attrs`.

It's a tough balance between supporting all usage explicitly (e.g. `url` added does variant A or B) but I opted for this approach as it would have made the template tag more complicated if we needed a `open_in_new_tab` type option.